File: 01jupyter/nb/source/cs01/include/omp_parallel_for.c
#include <stdio.h>
#include <stdlib.h>
#include <omp.h>

int main(int argc, char ** argv) {
  int m = (1 < argc ? atoi(argv[1]) : 10);
  printf("before parallel\n");
#pragma omp parallel
  {
    printf("in parallel, before for\n");
#pragma omp for
    for (int i = 0; i < m; i++) {
      printf("i = %3d\n", i);
    }
    printf("in parallel, after for\n");
  }
  printf("after parallel\n");
  return 0;
}

File: 01jupyter/nb/source/03_for_collapse/include/omp_parallel_for.f90
program omp_parallel_for
  use omp_lib
  character(len=32) :: arg
  integer :: m, i
  if (command_argument_count() >= 1) then
     call get_command_argument(1, arg)
     read (arg, *) m
  else
     m = 10
  end if
  print "(a)", "before parallel"
  !$omp parallel
  print "(a)", "in parallel, before do"
  !$omp do
  do i = 0, m - 1
     print "(a,i3)", "i = ", i
  end do
  !$omp end do
  print "(a)", "in parallel, after do"
  !$omp end parallel
  print "(a)", "after parallel"
end program omp_parallel_for
